diff -Naur ltp-full-20050804/testcases/kernel/syscalls/setregid/setregid02.c ltp-full-20050804-new/testcases/kernel/syscalls/setregid/setregid02.c
--- ltp-full-20050804/testcases/kernel/syscalls/setregid/setregid02.c	2005-08-04 13:33:06.000000000 -0700
+++ ltp-full-20050804-new/testcases/kernel/syscalls/setregid/setregid02.c	2005-08-17 11:33:35.000000000 -0700
@@ -106,8 +106,8 @@
 	{ &bin_gr_gid, &neg_one, EPERM, &users, &users, "After setregid(bin, -1)," },
 	{ &root_gr_gid, &bin_gr_gid, EPERM, &users, &users, "After setregid(root, bin)" },
 	{ &bin_gr_gid, &root_gr_gid, EPERM, &users, &users, "After setregid(bin, root)," },
-	{ &inval_user, &neg_one, EINVAL, &users, &users, "After setregid(invalid group, -1)," },
-	{ &neg_one, &inval_user, EINVAL, &users, &users, "After setregid(-1, invalid group)," },
+	{ &inval_user, &neg_one, EPERM, &users, &users, "After setregid(invalid group, -1)," },
+	{ &neg_one, &inval_user, EPERM, &users, &users, "After setregid(-1, invalid group)," },
 };
 
 int TST_TOTAL = sizeof(test_data)/sizeof(test_data[0]);
